serialize/src/impls.rs

```diff
@@ -147,6 +148,50 @@ impl CanonicalDeserialize for usize {
     }
 }
 
+impl CanonicalSerialize for BigUint {
+    #[inline]
+    fn serialize_with_mode<W: Write>(
+        &self,
+        writer: W,
+        compress: Compress,
+    ) -> Result<(), SerializationError> {
+        self.to_bytes_le().serialize_with_mode(writer, compress)
```

Comment:
We should check that this serializes the length (maybe putting a regression test?)